chat-commands.js

```diff
@@ -2503,47 +2512,106 @@ exports.commands = {
 				// broadcast the new formats list to clients
 				Rooms.global.send(Rooms.global.formatListText);
 
-				return this.sendReply("Formats have been hotpatched.");
+				this.sendReply("Formats have been hot-patched.");
 			} else if (target === 'loginserver') {
 				FS('config/custom.css').unwatch();
 				Chat.uncacheTree('./loginserver');
 				global.LoginServer = require('./loginserver');
-				return this.sendReply("The login server has been hotpatched. New login server requests will use the new code.");
+				this.sendReply("The login server has been hot-patched. New login server requests will use the new code.");
 			} else if (target === 'learnsets' || target === 'validator') {
+				if (hotpatchLock['validator']) return this.errorReply(`Hot-patching the validator has been disabled by ${hotpatchLock['validator'].by} (${hotpatchLock['validator'].reason})`);
+				if (hotpatchLock['formats']) return this.errorReply(`Hot-patching formats has been disabled by ${hotpatchLock['formats'].by} (${hotpatchLock['formats'].reason})`);
+
 				TeamValidator.PM.respawn();
-				return this.sendReply("The team validator has been hotpatched. Any battles started after now will have teams be validated according to the new code.");
+				this.sendReply("The team validator has been hot-patched. Any battles started after now will have teams be validated according to the new code.");
 			} else if (target === 'punishments') {
 				delete require.cache[require.resolve('./punishments')];
 				global.Punishments = require('./punishments');
-				return this.sendReply("Punishments have been hotpatched.");
+				this.sendReply("Punishments have been hot-patched.");
 			} else if (target === 'dnsbl' || target === 'datacenters') {
 				Dnsbl.loadDatacenters();
-				return this.sendReply("Dnsbl has been hotpatched.");
-			} else if (target.startsWith('disablechat')) {
-				if (Monitor.hotpatchLockChat) return this.errorReply("Hotpatch is already disabled.");
-				let reason = target.split(', ')[1];
-				if (!reason) return this.errorReply("Usage: /hotpatch disablechat, [reason]");
-				Monitor.hotpatchLockChat = reason;
-				return this.sendReply("You have disabled hotpatch until the next server restart.");
+				this.sendReply("Dnsbl has been hot-patched.");
 			} else if (target.startsWith('disable')) {
-				let reason = target.split(', ')[1];
-				if (!reason) return this.errorReply("Usage: /hotpatch disable, [reason]");
-				Monitor.hotpatchLock = reason;
-				return this.sendReply("You have disabled hotpatch until the next server restart.");
+				this.sendReply("Disabling hot-patch has been moved to its' own command:");
```

Comment:
Done

(on a side-note, I really need to practice when and where to use apostrophes, lol)